packages/calypso-e2e/src/rest-api-client.ts

```diff
@@ -635,6 +636,33 @@ export class RestAPIClient {
 
 	/* Posts */
 
+	/**
+	 * Given a siteID, checks whether any posts exists of a given state.
+	 *
+	 * @param {number} siteID Site ID.
+	 * @param param1 Keyed object parameter.
+	 * @param {SitePostState} param1.state State of the published post.
+	 */
+	async siteHasPost(
```

Comment:
**suggestion (non-blocking):** We can also check for the expected response structure & API-specific errors. 

```
if ( response.hasOwnProperty( 'error' ) ) {
	throw new Error(
		`${ ( response as ErrorResponse ).error }: ${ ( response as ErrorResponse ).message }`
	);
}

if (!response || !response.counts || !response.counts.all) {
	throw new Error('Unexpected response structure');
}
```

Reply:
So, I played around with this a bit, and it appears this endpoint doesn't ever return an error - it only returns a 404 (if _anything_ in the path is wrong, for example site ID) or the actual values.